src/main/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource.java

```diff
@@ -282,17 +285,22 @@ public CommunityOrPrisonOffenderManager switchResponsibleOfficer(final @PathVari
     })
     @ApiOperation(value = "Returns the offender summary for the given crn", tags = "-- Popular core APIs --")
 
-    public OffenderDetailSummary getOffenderSummaryByCrn(final @PathVariable("crn") String crn) {
-        final var offender = offenderService.getOffenderSummaryByCrn(crn);
-        return offender.orElseThrow(() -> new NotFoundException(String.format("Offender with crn %s not found", crn)));
+    public OffenderDetailSummary getOffenderSummaryByCrn(final @PathVariable("crn") String crn, Authentication authentication) {
+        userAccessService.checkExclusionsAndRestrictions(crn, authentication.getAuthorities());
+
+        return offenderService.getOffenderSummaryByCrn(crn)
+            .orElseThrow(() -> new NotFoundException(String.format("Offender with crn %s not found", crn)));
     }
 
     @RequestMapping(value = "/offenders/crn/{crn}/all", method = RequestMethod.GET)
     @ApiResponses(value = {
-            @ApiResponse(code = 404, message = "The offender is not found")
+            @ApiResponse(code = 404, message = "The offender is not found"),
+            @ApiResponse(code = 403, message = "Forbidden, the offender may have exclusions or restrictions in place preventing some users from viewing. Adopting client role ROLE_COMMUNITY_API_OPEN can bypass these restrictions."),
     })
     @ApiOperation(value = "Returns the full offender detail for the given crn", tags = "-- Popular core APIs --")
-    public OffenderDetail getOffenderDetailByCrn(final @PathVariable("crn") String crn) {
+    public OffenderDetail getOffenderDetailByCrn(final @PathVariable("crn") String crn, Authentication authentication) {
+        userAccessService.checkExclusionsAndRestrictions(crn, authentication.getAuthorities());
```

Comment:
Interestingly other projects does this slightly differently by injecting a SecurityUserContext where you can retrieve the Authentication, but this works just as well IMHO

Reply:
Yeah I saw that and did consider updating it but clearly never got round to it happy to do it either way!